internal/provider/providers/porkbun/provider.go

```diff
@@ -119,16 +119,31 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 		recordType = constants.AAAA
 	}
 	ipStr := ip.String()
-	recordIDs, err := p.getRecordIDs(ctx, client, recordType)
+	records, err := p.getRecords(ctx, client, recordType)
 	if err != nil {
 		return netip.Addr{}, fmt.Errorf("getting record IDs: %w", err)
 	}
 
-	if len(recordIDs) == 0 {
-		// ALIAS record needs to be deleted to allow creating an A record.
-		err = p.deleteALIASRecordIfNeeded(ctx, client)
-		if err != nil {
-			return netip.Addr{}, fmt.Errorf("deleting ALIAS record if needed: %w", err)
+	if len(records) == 0 {
+		// For new domains, Porkbun Creates 2 Default DNS Records which point to their "parked" domain page:
+		// ALIAS domain.tld -> pixie.porkbun.com
+		// CNAME *.domain.tld -> pixie.porkbun.com
+		// ALIAS and CNAME records conflict with A and AAAA records, and attempting to create an A or AAAA record
+		// will return a 400 error if they aren't first deleted.
+		porkbunParkedDomain := "pixie.porkbun.com"
+		switch {
+		case p.owner == "@":
+			// Delete ALIAS domain.tld -> pixie.porkbun.com record
+			err = p.deleteMatchingRecord(ctx, client, constants.ALIAS, porkbunParkedDomain)
+			if err != nil {
+				return netip.Addr{}, err
+			}
+		case p.owner == "*":
+			// Delete CNAME *.domain.tld -> pixie.porkbun.com record
```

Comment:
Hmmmm, so I do agree, but it shouldn't error out if there IS an alias that's set to something else... right?

Additionally, as it's currently coded we use the `p.owner` to get records, so will probably need to start passing in what domain we want to fetch / delete for instead of using the current `p.owner` does that make sense / seem like the direction you want to go? would certainly make the API less tied to the context of the provider.